src/util/amount.rs

```diff
@@ -69,12 +69,31 @@ impl FromStr for Denomination {
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         match s {
             "BTC" => Ok(Denomination::Bitcoin),
+            "btc" => Ok(Denomination::Bitcoin),
             "mBTC" => Ok(Denomination::MilliBitcoin),
+            "mbtc" => Ok(Denomination::MilliBitcoin),
             "uBTC" => Ok(Denomination::MicroBitcoin),
+            "ubtc" => Ok(Denomination::MicroBitcoin),
+            "Bit" => Ok(Denomination::Bit),
+            "bit" => Ok(Denomination::Bit),
+            "Bits" => Ok(Denomination::Bit),
             "bits" => Ok(Denomination::Bit),
+            "Satoshi" => Ok(Denomination::Satoshi),
             "satoshi" => Ok(Denomination::Satoshi),
+            "Satoshis" => Ok(Denomination::Satoshi),
+            "satoshis" => Ok(Denomination::Satoshi),
+            "SAT" => Ok(Denomination::Satoshi),
+            "Sat" => Ok(Denomination::Satoshi),
             "sat" => Ok(Denomination::Satoshi),
+            "SATS" => Ok(Denomination::Satoshi),
+            "Sats" => Ok(Denomination::Satoshi),
+            "sats" => Ok(Denomination::Satoshi),
+            "mSAT" => Ok(Denomination::MilliSatoshi),
+            "mSat" => Ok(Denomination::MilliSatoshi),
             "msat" => Ok(Denomination::MilliSatoshi),
+            "mSATS" => Ok(Denomination::MilliSatoshi),
+            "mSats" => Ok(Denomination::MilliSatoshi),
+            "msats" => Ok(Denomination::MilliSatoshi),
             d => Err(ParseAmountError::UnknownDenomination(d.to_owned())),
```

Comment:
I think ifs with [`eq_ignore_ascii_case`](https://doc.rust-lang.org/std/primitive.str.html#method.eq_ignore_ascii_case) would be better - more concise and obvious what it does. It would also allow ridiculous strings like `10 sAtS` but I don't see harm in it.

Reply:
Maybe we could forbid `Msat` (any sats starting with capital `M`) as suspicious - possibly meaning "mega sats"

Reply:
I ended up squashing both commits and going with 
```
        // "sat", "sats", whatever.
        let s = s.strip_suffix('s').unwrap_or(s);

        if s == "Msat" {        // So as not to confuse with mega sats.
            return Err(ParseAmountError::UnknownDenomination(s.to_owned()));
        }

        if s.eq_ignore_ascii_case("BTC") {
            return Ok(Denomination::Bitcoin)
        }
        ...
```

Reply:
Done. In case you are wondering what all the force pushing is about. I removed the suffix 's' but didn't test locally with 1.29 so it was no good :(

The code is pretty ugly.